src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.GeneratedFileReplacingCompilationTracker.cs

```diff
@@ -122,15 +106,14 @@ public async Task<Compilation> GetCompilationAsync(SolutionState solution, Cance
                 return _compilationWithReplacement;
             }
 
+            public Task<VersionStamp> GetDependentVersionAsync(SolutionState solution, CancellationToken cancellationToken)
+                => _underlyingTracker.GetDependentVersionAsync(solution, cancellationToken);
+
             public Task<VersionStamp> GetDependentSemanticVersionAsync(SolutionState solution, CancellationToken cancellationToken)
-            {
-                return _underlyingTracker.GetDependentSemanticVersionAsync(solution, cancellationToken);
-            }
+                => _underlyingTracker.GetDependentSemanticVersionAsync(solution, cancellationToken);
 
-            public Task<VersionStamp> GetDependentVersionAsync(SolutionState solution, CancellationToken cancellationToken)
-            {
-                return _underlyingTracker.GetDependentVersionAsync(solution, cancellationToken);
-            }
+            public Task<Checksum> GetDependentChecksumAsync(SolutionState solution, CancellationToken cancellationToken)
+                => _underlyingTracker.GetDependentChecksumAsync(solution, cancellationToken);
```

Comment:
This isn't quite right. We're going to have a GeneratedFileReplacingCompilationTracker if we have a stale generated file in the IDE that hasn't been updated yet due to async updating. The case is the snapshot has moved forward, generated outputs _will_ change, but the user invokes the feature on a stale file and we need spans to line up. Since we're sticking a different file in, conceivably we would have different semantics and we should probably have a different checksum here. Combining the checksum of _underlyingTracker and _replacedGeneratedDocumentState (which should have a checksum like any other file) would probably be sufficient.

We only will ever have one of these compilation trackers in a forked case (i.e. it'll never exist in a Workspace's CurrentSolution), and I don't think it's actually going to be the case where it's important for us to have cached results for this. But I'd be worried if we ended up saving/loading cached results when it's technically different. Put another way, if you replaced this implementation with "generate a new GUID and checksum that" it'd probably be observably the same and would avoid accidental cache pollution or bad cached results.